qiskit_experiments/randomized_benchmarking/rb_utils.py

```diff
@@ -0,0 +1,262 @@
+        """
+        error_dict = {}
+        for backend_gate in backend.properties().gates:
+            backend_gate = backend_gate.to_dict()
+            gate_qubits = tuple(backend_gate["qubits"])
+            if all(gate_qubit in qubits for gate_qubit in gate_qubits):
+                for p in backend_gate["parameters"]:
+                    if p["name"] == "gate_error":
+                        error_dict[(gate_qubits, backend_gate["gate"])] = p["value"]
+        return error_dict
+
+    @staticmethod
+    def count_ops(
+        circuit: QuantumCircuit, qubits: Optional[Iterable[int]] = None
+    ) -> Dict[Tuple[Iterable[int], str], int]:
+        """Counts occurances of each gate in the given circuit
+
+        Args:
+            circuit: The quantum circuit whose gates are counted
+            qubits: A list of qubits to filter irrelevant gates
+
+        Returns:
+            A dictionary of the form (qubits, gate) -> value where value
+            is the number of occurrences of the gate on the given qubits
+        """
+        if qubits is None:
+            qubits = range(len(circuit.qubits))
+        count_ops_result = {}
+        for instr, qargs, _ in circuit._data:
+            instr_qubits = []
+            skip_instr = False
+            for qubit in qargs:
+                qubit_index = circuit.qubits.index(qubit)
+                if qubit_index not in qubits:
+                    skip_instr = True
+                instr_qubits.append(qubit_index)
+            if not skip_instr:
+                instr_qubits = tuple(instr_qubits)
+                count_ops_result[(instr_qubits, instr.name)] = (
+                    count_ops_result.get((instr_qubits, instr.name), 0) + 1
+                )
+        return count_ops_result
+
+    @staticmethod
+    def gates_per_clifford(
+        ops_count: List,
+    ) -> Dict[Tuple[Iterable[int], str], float]:
+        """
+        Computes the average number of gates per clifford for each gate type
+        in the input from raw count data coming from multiple circuits.
+        Args:
+            ops_count: A List of [key, value] pairs where
+            key is [qubits, gate_name] and value is the average
+            number of gates per clifford of the type for the given key
+
+        Returns:
+            A dictionary with the mean value of values corresponding
+            to key for each key.
+
+        """
+        result = {}
+        for ((qubits, gate_name), value) in ops_count:
+            qubits = tuple(qubits)  # so we can hash
+            if (qubits, gate_name) not in result:
+                result[(qubits, gate_name)] = []
+            result[(qubits, gate_name)].append(value)
+        return {key: np.mean(value) for (key, value) in result.items()}
+
+    @staticmethod
+    def coherence_limit(nQ=2, T1_list=None, T2_list=None, gatelen=0.1):
+        """
+        The error per gate (1-average_gate_fidelity) given by the T1,T2 limit.
```

Comment:
Can you write physics behind this? This is quite convenient but I felt some difficulty of using this in journal parpers, because we don't provide any technical reference for this calculation.

Reply:
Note that this code was copied from the ignis code written by Dave, so you should refer this question to him:
https://github.com/Qiskit/qiskit-ignis/blob/9201ed8f9970c50308601f4dfd895a1b8255e469/qiskit/ignis/verification/randomized_benchmarking/rb_utils.py#L169-#L224

Reply:
This is not originally my code, but I'll try.

Reply:
Actually I don't know the original paper of this calculation. Seems like some conventional code used by IBM experimentalists. One thing really close to this form is shown in Appendix G of [this paper](https://arxiv.org/pdf/2007.02925.pdf).

It would be great if you can derive the coherence limit, but this is not mandatory for this PR. Can you write an issue for future extension?